packages/@react-spectrum/table/test/Table.test.js

```diff
@@ -2618,28 +2660,65 @@ describe('TableView', function () {
+        expect(announce).toHaveBeenLastCalledWith('Foo 5 selected.');
+        expect(announce).toHaveBeenCalledTimes(1);
         checkSelection(onSelectionChange, ['Foo 5']);
 
+        announce.mockReset();
         onSelectionChange.mockReset();
         fireEvent.keyDown(document.activeElement, {key: 'ArrowDown', ctrlKey: true});
         fireEvent.keyUp(document.activeElement, {key: 'ArrowDown', ctrlKey: true});
+        expect(announce).not.toHaveBeenCalled();
         expect(onSelectionChange).not.toHaveBeenCalled();
         expect(document.activeElement).toBe(getCell(tree, 'Baz 6').closest('[role="row"]'));
 
         fireEvent.keyDown(document.activeElement, {key: 'ArrowDown', ctrlKey: true});
         fireEvent.keyUp(document.activeElement, {key: 'ArrowDown', ctrlKey: true});
+        expect(announce).not.toHaveBeenCalled();
         expect(onSelectionChange).not.toHaveBeenCalled();
         expect(document.activeElement).toBe(getCell(tree, 'Baz 7').closest('[role="row"]'));
 
         fireEvent.keyDown(document.activeElement, {key: ' ', ctrlKey: true});
         fireEvent.keyUp(document.activeElement, {key: ' ', ctrlKey: true});
+        expect(announce).toHaveBeenCalledWith('Foo 7 selected. 2 items selected.');
+        expect(announce).toHaveBeenCalledTimes(1);
         checkSelection(onSelectionChange, ['Foo 5', 'Foo 7']);
 
+        announce.mockReset();
         onSelectionChange.mockReset();
         fireEvent.keyDown(document.activeElement, {key: ' '});
         fireEvent.keyUp(document.activeElement, {key: ' '});
+        expect(announce).toHaveBeenCalledWith('Foo 5 not selected. 1 item selected.');
+        expect(announce).toHaveBeenCalledTimes(1);
         checkSelection(onSelectionChange, ['Foo 7']);
       });
+
+      it('should support non-contiguous selection with the keyboard', function () {
+        let onSelectionChange = jest.fn();
+        let tree = renderTable({onSelectionChange, selectionStyle: 'highlight'}, items);
+
+        userEvent.click(getCell(tree, 'Baz 1'));
+        expect(announce).toHaveBeenLastCalledWith('Foo 1 selected.');
+        expect(announce).toHaveBeenCalledTimes(1);
+        checkSelection(onSelectionChange, ['Foo 1']);
+
+        announce.mockReset();
+        onSelectionChange.mockReset();
+        fireEvent.keyDown(document.activeElement, {key: 'ArrowDown'});
+        fireEvent.keyUp(document.activeElement, {key: 'ArrowDown'});
+        expect(announce).not.toHaveBeenCalled();
+        checkSelection(onSelectionChange, ['Foo 2']);
+
+        onSelectionChange.mockReset();
+        fireEvent.keyDown(document.activeElement, {key: 'ArrowDown'});
+        fireEvent.keyUp(document.activeElement, {key: 'ArrowDown'});
+        expect(announce).not.toHaveBeenCalled();
+        expect(onSelectionChange).not.toHaveBeenCalled();
+
+        userEvent.click(getCell(tree, 'Baz 1'));
+        expect(announce).toHaveBeenCalledTimes(1);
+        checkSelection(onSelectionChange, ['Foo 1']);
```

Comment:
Two questions about this, if we are using keyboard, why click instead of enter/space to select? Shouldn't the selectionChange be two items and clicking be on a different item then the first click?